logic/standalone/waterfall.py

```diff
@@ -0,0 +1,188 @@
+'''
+Logic module that can create waterfall objects.
+
+
+USAGE EXAMPLE:
+
+'''
+
+import logic.common.log_utils as log
+import logic.common.tiled_utils as tiled_utils
+
+#--------------------------------------------------#
+'''Variables'''
+
+FG_PREFIX = "fg"		# Prefix of fg tilelayers
+BG_PREFIX = "bg"		# Prefix of bg tilelayers
+BLOCKOUT_PREFIX = "_FALLS_"	# Prefix of tilelayers with waterfall blockout
+OBJECTGROUP_PREFIX = "objects_waterfall_"
+
+LAYER_SORT_DIFF = 10	# Difference of sort_value between consecutive tilelayers
+LAYER_SORT_ADDON = 1	# Add to sort_value
+
+
+
+
+
+#--------------------------------------------------#
+'''Public Functions'''
+
+def ScanForWaterFalls(playdo):
+	'''
+	For each applicable tilelayer:
+	 - Check layer name for sort_value used
+	 - Check layer name for template used
+	  e.g. _FALLS_meadow => uses either the "meadow bg" or "meadow fg" template object
+	 - Process the tile2D for waterfall coordinates
+
+	Then return a list of tuples that contain all of the above informations
+	  Each tuple represents a new waterfall object layer to be added
+	'''
+
+	''' =====[Pseudo-code]=====
+
+	list_waterfall_layer = []
```

Comment:
not a fan of the name "list_waterfall_layer", since what it is becomes confusing. Is it a layer? Is it a list?

Let's use the name "list_waterfalls_to_create"

Reply:
How about just `list_all_waterfalls`? I kinda want to avoid starting a variable with `list_waterfall_something` since it makes searching (and replacing) slightly inconvenient.